doc/source/ray-air/examples/opt_deepspeed_inference.ipynb

```diff
@@ -0,0 +1,913 @@
+    "    runtime_env={\n",
+    "        \"pip\": [\n",
+    "            \"numpy==1.23\",\n",
+    "            \"protobuf==3.20.0\",\n",
+    "            \"transformers==4.27.2\",\n",
+    "            \"accelerate==0.17.1\",\n",
+    "            \"deepspeed==0.8.3\",\n",
+    "        ],\n",
+    "        \"env_vars\": {\n",
+    "            \"HF_HUB_DISABLE_PROGRESS_BARS\": \"1\",\n",
+    "        }\n",
+    "    }\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "613df744",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# Configurations and Hyperparameters\n",
+    "\n",
+    "from dataclasses import dataclass\n",
+    "\n",
+    "\n",
+    "# Following configuration tested for a single p3.16xlarge instance with 16 V100 GPUs.\n",
+    "\n",
+    "@dataclass\n",
+    "class Config:\n",
+    "    \"\"\"See https://deepspeed.readthedocs.io/en/stable/inference-init.html#deepspeed.inference.config.DeepSpeedInferenceConfig\n",
+    "    for more details about the configurations.\n",
+    "    \"\"\"\n",
+    "    model_name: str = \"facebook/opt-30b\"\n",
+    "    # Path to HuggingFace cache directory. Default is ~/.cache/huggingface/.\n",
+    "    cache_dir: str = None\n",
+    "    # Path to the directory that actually holds model files.\n",
+    "    # E.g., ~/.cache/huggingface/models--facebook--opt-30b/snapshots/xxx/\n",
+    "    # If this path is not None, we will skip download models from HuggingFace.\n",
+    "    repo_root: str = None\n",
+    "    # This is essentially how many DeepSpeed-inference replicas to run for\n",
+    "    # this batch inference job.\n",
+    "    num_worker_groups: int = 1\n",
+    "    # Number of DeepSpeed workers per group.\n",
+    "    num_workers_per_group: int = 8\n",
+    "\n",
+    "    batch_size: int = 1\n",
+    "    dtype: str = \"float16\"\n",
+    "    # Maximum number of tokens DeepSpeed inference-engine can work with,\n",
+    "    # including the input and output tokens.\n",
+    "    max_tokens: int = 1024\n",
+    "    # Use meta tensors to initialize model.\n",
+    "    use_meta_tensor: bool = True\n",
+    "    # Use cache for generation.\n",
+    "    use_cache: bool = True\n",
+    "    # The path for which we want to save the loaded model with a checkpoint.\n",
+    "    save_mp_checkpoint_path: str = None\n",
+    "\n",
+    "\n",
+    "config = Config()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "63b8a84d-57a6-4430-8fe8-9811760b8b7c",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Caching model locally ...\n",
+      "Done. Model saved in /home/ray/.cache/huggingface/hub/models--facebook--opt-30b/snapshots/ceea0a90ac0f6fae7c2c34bcb40477438c152546\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Download and cache model on all GPU instances of the cluster.\n",
```

Comment:
If we are just downloading from hub, we probably don't need to call this ahead of time, as it will be automatically downloaded and cached by Hugging Face when creating the model object.

Reply:
Right. 2 considerations:
1. it felt a lot better to me to spread out the latency a little bit, and be explicit about the delay it needs to download the model. Otherwise it just looks like the Dataset pipeline hangs. I also had to disable HF's progress bar since it looks awful in notebook.
2. I added a comment so people can use this chance to copy the model files from s3 or network storage if they want.
let me know what you think.